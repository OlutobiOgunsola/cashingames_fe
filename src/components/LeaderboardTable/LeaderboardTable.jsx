import React, { useEffect, useState } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import TableItem from '../../widgets/TableItem'

const LeaderboardTable = (props) => {
    const data = props.data
    useEffect(() => {
        console.log(data)
    }, [data])
    const headers = ['POSITION', 'NAME', 'POINT']
    return <TableItem data={data} headers={headers} />
}

LeaderboardTable.propTypes = {
    data: PropTypes.array.isRequired,
}

export default LeaderboardTable
