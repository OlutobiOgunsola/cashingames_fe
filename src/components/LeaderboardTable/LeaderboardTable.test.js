import { mount } from 'enzyme'
import LeaderboardTable from './LeaderboardTable'
import toJSON from 'enzyme-to-json'

describe('Leaderboard component', () => {
    it('renders without crashing', () => {
        const data = []
        const headers = []
        const LWrapper = mount(
            <LeaderboardTable data={data} headers={headers} />
        )

        expect(toJSON(LWrapper)).toMatchSnapshot()
    })
    it('renders a table', () => {
        const data = [
            {
                id: 1,
                name: 'Taye',
            },
            {
                id: 2,
                name: 'Kehinde',
            },
        ]

        const headers = ['ID', 'NAME']

        const LWrapper = mount(
            <LeaderboardTable data={data} headers={headers} />
        )

        expect(LWrapper.find('#table-container')).toBeTruthy()
    })
    it('should render child with data props', () => {})
})
