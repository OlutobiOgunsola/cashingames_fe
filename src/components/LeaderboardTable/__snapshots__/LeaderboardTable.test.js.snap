// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard component renders without crashing 1`] = `
<LeaderboardTable
  data={Array []}
  headers={Array []}
>
  <TableItem
    data={Array []}
    headers={
      Array [
        "POSITION",
        "NAME",
        "POINT",
      ]
    }
  >
    <styled.div
      id="table-container"
    >
      <div
        className="sc-bdfBwQ eA-dDSP"
        id="table-container"
      >
        <ForwardRef
          hover={true}
        >
          <table
            className="table table-hover"
          >
            <thead>
              <tr>
                <th
                  className="table-header"
                  key="0"
                >
                  POSITION
                </th>
                <th
                  className="table-header"
                  key="1"
                >
                  NAME
                </th>
                <th
                  className="table-header"
                  key="2"
                >
                  POINT
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </ForwardRef>
      </div>
    </styled.div>
  </TableItem>
</LeaderboardTable>
`;
